comments --- Modules/_ctypes/malloc_closure.c
comments +++ Modules/_ctypes/malloc_closure.c
comments @@ -87,7 +87,7 @@
comments  }
comments  
comments  /******************************************************************/
comments -
comments +#if 0
comments  /* put the item back into the free list */
comments  void ffi_closure_free(void *p)
comments  {
comments @@ -114,3 +114,4 @@
comments  #endif
comments      return (void *)item;
comments  }
comments +#endif
--- Modules/_ctypes/cfield.c
+++ Modules/_ctypes/cfield.c
@@ -1510,7 +1510,11 @@
     return PyLong_FromVoidPtr(*(void **)ptr);
 }
 
-static struct fielddesc formattable[] = {
+#define FORMATTABLE_SIZE 30
+static struct fielddesc formattable[FORMATTABLE_SIZE];
+
+static void formattable_init(void) {
+struct fielddesc my_formattable[] = {
     { 's', s_set, s_get, &ffi_type_pointer},
     { 'b', b_set, b_get, &ffi_type_schar},
     { 'B', B_set, B_get, &ffi_type_uchar},
@@ -1564,6 +1568,11 @@
     { 'O', O_set, O_get, &ffi_type_pointer},
     { 0, NULL, NULL, NULL},
 };
+    size_t nb = 1;
+    for (struct fielddesc *pos = my_formattable; pos->code; ++pos, ++nb);
+    if (FORMATTABLE_SIZE < nb) abort();
+    memcpy(formattable, my_formattable, nb * sizeof(struct fielddesc));
+}
 
 /*
   Ideas: Implement VARIANT in this table, using 'V' code.
@@ -1578,6 +1587,7 @@
 
     if (!initialized) {
         initialized = 1;
+        formattable_init();
 #ifdef CTYPES_UNICODE
         if (sizeof(wchar_t) == sizeof(short))
             _ctypes_get_fielddesc("u")->pffi_type = &ffi_type_sshort;
@@ -1640,7 +1650,7 @@
     struct _ffi_type **elements;
 } ffi_type;
 */
-
+#if 0
 /* align and size are bogus for void, but they must not be zero */
 ffi_type ffi_type_void = { 1, 1, FFI_TYPE_VOID };
 
@@ -1667,5 +1677,5 @@
                                  FFI_TYPE_LONGDOUBLE };
 
 ffi_type ffi_type_pointer = { sizeof(void *), VOID_P_ALIGN, FFI_TYPE_POINTER };
-
+#endif
 /*---------------- EOF ----------------*/
